tensorrt_llm/executor/result.py

```diff
@@ -90,7 +107,7 @@ def token_ids_diff(self) -> List[int]:
         return self.token_ids[self._last_token_ids_len:]
 
     @property
-    def logprobs_diff(self) -> List[float]:
+    def logprobs_diff(self) -> List[dict[int, Logprob]]:
```

Comment:
We can unify all the `List/Dict` to `list/dict` as they are suggested from [python 3.9](https://docs.python.org/3.9/library/typing.html#typing.List)